src/inmanta/agent/handler.py

```diff
@@ -141,6 +141,16 @@ def actual(f: Callable[..., object]) -> T_FUNC:
         myignore = set(ignore)
         sig = inspect.signature(f)
         myargs = list(sig.parameters.keys())[1:]  # Starts at 1 because 0 is self.
+        if evict_after_creation > 0 and timeout and timeout > 0:
+            LOGGER.warning(
+                "Both the `evict_after_creation` and the deprecated `timeout` parameter are set "
+                "for cached method %s. The `timeout` parameter will be ignored and cached entries will "
+                "be kept in the cache for %.2fs after entering it. The `timeout` parameter should no"
+                "longer be used. Please refer to the handler documentation "
+                "for more information about setting a retention policy.",
+                f.__name__,
+                evict_after_creation,
+            )
```

Comment:
This warning can be triggered on an Inmanta project that consists of many modules. Like that it might not be sufficient to log the function name only. Also adding the module to it will help the user to diagnose which function in which module is causing the warning.
```suggestion
            LOGGER.warning(
                "Both the `evict_after_creation` and the deprecated `timeout` parameter are set "
                "for cached method %s.%s. The `timeout` parameter will be ignored and cached entries will "
                "be kept in the cache for %.2fs after entering it. The `timeout` parameter should no "
                "longer be used. Please refer to the handler documentation "
                "for more information about setting a retention policy.",
                f.__module__,
                f.__name__,
                evict_after_creation,
            )
```